hibernate-core/src/main/java/org/hibernate/sql/results/graph/entity/internal/BatchEntitySelectFetchInitializer.java

```diff
@@ -40,7 +40,7 @@ public BatchEntitySelectFetchInitializer(
 	protected void registerResolutionListener() {
 		parentAccess.registerResolutionListener( parentInstance -> {
 			final AttributeMapping parentAttribute = getParentEntityAttribute( referencedModelPart.getAttributeName() );
-			if ( parentAttribute != null ) {
+			if ( parentAttribute != null && !firstEntityInitializer.isEntityInitialized() ) {
```

Comment:
Can you explain the need for this condition please?

Reply:
if the parent entity is initialized, there is no need to add it to the batch loading.

Reply:
Wouldn't we skip running the initializer in this case anyway because the entity would have already been loaded?

Reply:
no, we don't